src/model/ValveTanh.cpp

```diff
@@ -56,4 +60,8 @@ void ValveTanh::update_solution(
   system.dC_dy.coeffRef(global_eqn_ids[0], global_var_ids[2]) =
       -0.5 * q_in * (Rmax - Rmin) * steep *
       (1.0 - tanh(steep * (p_out - p_in)) * tanh(steep * (p_out - p_in)));
+  system.dC_dy.coeffRef(global_eqn_ids[2], global_var_ids[0]) =
+      0.5 * steep / pow(cosh(steep * (p_in - p_out)), 2);
+  system.dC_dy.coeffRef(global_eqn_ids[2], global_var_ids[2]) =
+      -0.5 * steep / pow(cosh(steep * (p_in - p_out)), 2);
```

Comment:
Define a variable here as well since columns `0` and `2` only differ by their sign